sound/soc/sof/intel/hda.h

```diff
@@ -408,6 +408,9 @@ struct sof_intel_hda_dev {
 
 	int irq;
 
+	/* PM related */
+	bool l1_support_changed;/* during suspend, is L1SEN changed or not */
```

Comment:
why do you need an additional flag, you already have s0_suspend?

Reply:
No, s0_suspend can't meet the requirement here, comment in below.

Reply:
@keyonjie I have a suggestion. I think we should unconditionally enable L1 support when entering system suspend (both S3 and S0ix)

But upon resume, instead of doing the disabling in the resume callback, I think it makes sense to do it in the hda_dsp_pcm_trigger(). 

This means we can move the current code that enables/disables DMI L1 could also move to the trigger() instead of the open/close() callbacks. What do you think?

Reply:
I have another suggestion: how about we add this in a follow-up PR, this is not fully cooked and distracting us from the main parts of the PR

Reply:
> I have another suggestion: how about we add this in a follow-up PR, this is not fully cooked and distracting us from the main parts of the PR

But this is crucial for fixing the S0ix entry fail issue, without it, the whole KWD in S0ix feature doesn't work. :(

Reply:
> @keyonjie I have a suggestion. I think we should unconditionally enable L1 support when entering system suspend (both S3 and S0ix)
> 
> But upon resume, instead of doing the disabling in the resume callback, I think it makes sense to do it in the hda_dsp_pcm_trigger().
> 
> This means we can move the current code that enables/disables DMI L1 could also move to the trigger() instead of the open/close() callbacks. What do you think?

@ranj063 got your point, I originally preferred this solution, but then an fact defeated me: for the KWD capture and DMA trace capture, they are not triggered stop before entering S0Ix, that means, there will be no corresponding triggered stop for them on resuming, which means that the L1SEN is changed(from 0 ->1) with your solution, if they are the only capture streams opened before entering S0ix.

Reply:
> for the KWD capture and DMA trace capture, they are not triggered stop before entering S0Ix,

@keyonjie this is not correct. They are not "STOPPED" but they are "SUSPENDED" isnt it?